src/main/java/org/opensearch/security/support/Base64Helper.java

```diff
@@ -200,4 +308,66 @@ protected Class<?> resolveClass(ObjectStreamClass desc) throws IOException, Clas
             throw new InvalidClassException("Unauthorized deserialization attempt ", clazz.getName());
         }
     }
+
+    private static boolean isWriteable(Class<?> clazz) {
+        return Writeable.class.isAssignableFrom(clazz);
+    }
+
+    /**
+     * Returns integer ID for the registered Writeable class
+     * <br/>
+     * Protected for testing
+     */
+    protected static Integer getWriteableClassID(Class<?> clazz) {
+        if ( !isWriteable(clazz) ) {
+            throw new OpenSearchException("clazz should implement Writeable ", clazz);
+        }
+        if( !writeableClassToIdMap.containsKey(clazz) ) {
+            throw new OpenSearchException("Writeable clazz not registered ", clazz);
+        }
+        return writeableClassToIdMap.get(clazz);
+    }
+
+    private static Class<?> getWriteableClassFromId(int id) {
+        return writeableClassToIdMap.inverse().get(id);
+    }
+
+    /**
+     * Registers the given <code>Writeable</code> class for custom serialization by assigning an incrementing integer ID
+     * IDs are stored in a HashBiMap
+     * @param clazz class to be registered
+     */
+    private static void registerWriteable(Class<? extends Writeable> clazz) {
+        if ( writeableClassToIdMap.containsKey(clazz) ) {
+            throw new OpenSearchException("writeable clazz is already registered ", clazz.getName());
+        }
+        int id = writeableClassToIdMap.size() + 1;
+        writeableClassToIdMap.put(clazz, id);
+    }
+
+    /**
+     * Registers all <code>Writeable</code> classes for custom serialization support.
+     * Removing existing classes / changing order of registration will cause a breaking change in the serialization protocol
+     * as <code>registerWriteable</code> assigns an incrementing integer ID to each of the classes in the order it is called
+     * starting from <code>1</code>.
+     *<br/>
+     * New classes can safely be added towards the end.
```

Comment:
Thank you for this comment. Order matters in this list since the `ordinal` is used for an ID in a map.